waspc/packages/deploy/src/providers/fly/setup/setup.ts

```diff
@@ -0,0 +1,72 @@
+import { $, cd } from 'zx'
+import crypto from 'crypto'
+import * as tomlHelpers from '../helpers/tomlFileHelpers.js'
+import { createDeploymentInfo, DeploymentInfo } from '../DeploymentInfo.js'
+import { GlobalOptions } from '../GlobalOptions.js'
+import { buildDirExists, cdToClientBuildDir, cdToServerBuildDir, lazyInit, getCommandHelp, waspSays } from '../helpers/helpers.js'
+import { createFlyDbCommand } from '../index.js'
+
+export async function setup(baseName: string, region: string, options: GlobalOptions) {
+  waspSays('Setting up your Wasp app with Fly.io!')
+
+  const buildWaspIfMissing = lazyInit(async () => {
+    if (!buildDirExists(options.waspDir)) {
```

Comment:
We only need to infer the project type to setup the Fly apps, so it is not imperative that the build dir be fully up to date, but rather just "look like" a Wasp project. However, should we just error on the side of safety and force a `wasp build` here, just in case? 🤔 Hmm, probably. Note to @shayneczyzewski to do so.

Reply:
Ok so I made setup always build, just to make sure the build dir is in a good known state. If others think it should only build if not present (and thus perhaps stale) please let me know.

This means the normal full setup flow will now build twice (once for setup, once later for deploy, but you can skip build on deploy if you want via optional opts).

Reply:
Maybe `setup`, it if would in chain call other commands, could by default call the `deploy` with option to skip rebuilding?

Reply:
True! If we do the `launch` command that just runs all 3 sequentially, it could do setup (with build), create DB, then deploy (without build). I like it 👍🏻 thnx!